tests/rule_based_profiler/parameter_builder/test_parameter_builder.py

```diff
@@ -0,0 +1,259 @@
+)
+from great_expectations.rule_based_profiler.parameter_builder import (
+    ParameterBuilder,
+)
+from great_expectations.types.attributes import Attributes
+from great_expectations.rule_based_profiler.parameter_container import (
+    ParameterContainer,
+    ParameterNode,
+    FULLY_QUALIFIED_PARAMETER_NAME_VALUE_KEY,
+    FULLY_QUALIFIED_PARAMETER_NAME_ATTRIBUTED_VALUE_KEY,
+    FULLY_QUALIFIED_PARAMETER_NAME_METADATA_KEY,
+)
+
+
+class DummyParameterBuilder(ParameterBuilder):
+    def _build_parameters(
+        self,
+        domain: Domain,
+        variables: Optional[ParameterContainer] = None,
+        parameters: Optional[Dict[str, ParameterContainer]] = None,
+        runtime_configuration: Optional[dict] = None,
+    ) -> Attributes:
+        """
+        Builds ParameterContainer object that holds ParameterNode objects with attribute name-value pairs and details.
+
+        Returns:
+            Attributes object, containing computed parameter values and parameter computation details metadata.
+        """
+        return Attributes(
+            {
+                FULLY_QUALIFIED_PARAMETER_NAME_VALUE_KEY: 1,
+                FULLY_QUALIFIED_PARAMETER_NAME_ATTRIBUTED_VALUE_KEY: {"batch_id_0": 1},
+                FULLY_QUALIFIED_PARAMETER_NAME_METADATA_KEY: {"environment": "test"},
+            }
+        )
+
+
+@pytest.mark.unit
+@mock.patch("test_parameter_builder.DummyParameterBuilder._build_parameters")
+def test_parameter_builder_dependencies_evaluated_separately(
+    mock_parameter_builder__build_parameters: mock.MagicMock,
+):
+    my_evaluation_dependency_0_parameter_builder: ParameterBuilder = (
+        DummyParameterBuilder(
+            name="my_evaluation_dependency_parameter_name_0",
+            evaluation_parameter_builder_configs=None,
+            data_context=None,
+        )
+    )
+    my_evaluation_dependency_1_parameter_builder: ParameterBuilder = (
+        DummyParameterBuilder(
+            name="my_evaluation_dependency_parameter_name_1",
+            evaluation_parameter_builder_configs=None,
+            data_context=None,
+        )
+    )
+
+    my_dependent_parameter_builder: ParameterBuilder = DummyParameterBuilder(
+        name="my_dependent_parameter_builder",
+        evaluation_parameter_builder_configs=None,
+        data_context=None,
+    )
+
+    domain = Domain(
+        domain_type=MetricDomainTypes.TABLE,
+        domain_kwargs=None,
+        rule_name="my_rule",
+    )
+
+    variables: Optional[ParameterContainer] = None
+
+    parameter_container = ParameterContainer(parameter_nodes=None)
+    parameters: Dict[str, ParameterContainer] = {
+        domain.id: parameter_container,
+    }
+
+    my_evaluation_dependency_0_parameter_builder.build_parameters(
+        domain=domain,
+        variables=variables,
+        parameters=parameters,
+        batch_request=None,
+        runtime_configuration=None,
+    )
+    assert mock_parameter_builder__build_parameters.call_count == 1
```

Comment:
aren't you invoking your dummy parameter builder and then asserting that it was invoked? I fail to see what is being tested here

Reply:
@donaldheppner Please see `https://github.com/great-expectations/great_expectations/pull/8302#discussion_r1262574558` (we call the interface method, but test on the implementation).  Thanks.

Reply:
I don't really know if testing that a private method is called by a public interface is really providing much value - feels like we're testing an implementation detail instead of the behavior.

Reply:
@cdkini We have to test that the implementation method in the ParameterBuilder, listed as dependency, is called.  In the interface method, there is actually logic determining how and when to call the implementation method.  This is located around line 168 in `great_expectations/rule_based_profiler/parameter_builder/parameter_builder.py`.  Thanks.